### ISSUE TITLE
Cascade: pin only unobservable junction modes instead of a pinv cutoff

### ISSUE THREAD

**Issue by gvcallen** [gvcallen/paramrf#213]
## Parent

#173 (triage, closed in favour of this ticket)

## What to build

A scattering `Cascade` (and `RepeatedCascade`) of floating multi-port networks gives the same S-parameters and JAX gradients as the equivalent `Circuit.from_chain` build. Genuine near-singular behaviour, such as a lossless high-Q resonance, stays correct.

Today each junction inverts `M = I − B11·A22` (and `N = I − A22·B11`) with `pinv` at a user-set cutoff (`eps = 1e-12`). Floating networks leave a common mode at the junction that is singular in theory, but its measured σmin/σmax (3e-12 to 8e-12) sits just above that cutoff. `pinv` then inverts rounding noise: S is off by about 1e-5 and AD gradients are wrong by orders of magnitude (+2e8 against a correct −32.1). No fixed cutoff works: 1e-8 fixes floating chains but drops a real resonance next to a normal channel. See #173 for the full analysis.

The fix pins only junction directions that are **both singular and unobservable from the external ports**, and inverts everything else exactly. Unobservable means `A12 y = 0` for `M` and `B21 y = 0` for `N`. The cascade result doesn't depend on those directions, and a real resonance is observable, so it is never touched. The relative tolerance comes from the dtype, not the user. The SVD sits behind `stop_gradient`, so AD never differentiates through its zero singular values.

Decision-bearing snippet from the prototype (validated as a monkeypatch; the scripts are local in `notes/paramrf_cascade_fix/`):

```python
def _junction_inverse(M, out, rtol):
    K = jax.lax.stop_gradient(jnp.concatenate((M, out), axis=0))
    _, sv, Vh = jnp.linalg.svd(K, full_matrices=False)
    R = Vh.conj().T * (sv <= rtol * sv[0])      # singular AND unobservable
    return jnp.linalg.inv(M + R @ R.conj().T)

rtol = sqrt(finfo(dtype).eps)
X = _junction_inverse(I - B11 @ A22, A12, rtol)
Y = _junction_inverse(I - A22 @ B11, B21, rtol)
```

**Seam:** the cascade reduction in `pmrf.rf` (the two-network combine and the scattering reduction), the `Cascade` and `RepeatedCascade` models, and their tests.

### Decisions (from triage of #173)

1. **Remove `eps` outright.** It goes from `Cascade`, `RepeatedCascade` and the `pmrf.rf` cascade functions, with no shim or deprecation (pre-1.0). Nothing in the repo passes it explicitly beyond internal plumbing.
2. **Non-reciprocal models:** the fix's exactness argument assumes the left and right null spaces aren't orthogonal. That is shown for floating modes, not for non-reciprocal ones. Cover it with a regression test, not a separate analysis.
3. **Cost is accepted.** The extra SVD per junction makes a 20-block floating chain at 2001 points go from about 556 ms to about 760 ms. Don't add a fast path that brings back a cutoff.
4. **Known limit, document only:** a lossless model evaluated exactly at a true resonance frequency is still singular. That is physical and unchanged. Note it in the docstring.

### Testing notes

JAX gradients must be tested explicitly. Use the `Circuit.from_chain` gradient as the primary reference and central finite differences as a sanity check. With the current code, S carries noise at about the 1e-5 level, so finite differences alone can't tell which answer is right.

## Acceptance criteria

- [ ] A `Cascade` of floating lines, then a floating transition (`GroundExposed` joined to a `Tee`), then another floating line matches `Circuit.from_chain` in S. Target: about 1e-9. The prototype reaches 6.7e-10.
- [ ] **JAX gradient test (floating chain):** `jax.grad` of a fixed linear loss on S, with respect to a line length, matches the `Circuit.from_chain` gradient and agrees with finite differences.
- [ ] **Resonator test:** a lossless high-Q resonance (e.g. C = 1e-16) next to a normal channel in a `Cascade` stays correct at the resonance peak in S, and its JAX gradient matches `Circuit` and finite differences.
- [ ] A single two-port lossless resonator at resonance stays correct in S.
- [ ] **Non-reciprocal test:** a `Cascade` containing a non-reciprocal block next to a floating junction matches `Circuit.from_chain` in S and in JAX gradient.
- [ ] `RepeatedCascade` of floating sections matches the equivalent `Cascade` / `Circuit` in S and gradient.
- [ ] `eps` is gone from `Cascade`, `RepeatedCascade` and the `pmrf.rf` cascade functions. No tuned singular-value cutoff remains in the scattering cascade.
- [ ] The docstrings describe the unobservable-mode pinning and the lossless-at-resonance limit.
- [ ] The existing cascade tests pass, with no tolerance widened.

## Blocked by

None (can start immediately).
